dip-template/runtimes/dip-provider/src/lib.rs

```diff
@@ -568,14 +569,14 @@ impl_runtime_apis! {
 		}
 	}
 
-	impl kilt_runtime_api_dip_provider::DipProvider<Block, RuntimeApiDipProofRequest, CompleteMerkleProof<Hash, DidMerkleProofOf<Runtime>>, RuntimeApiDipProofError> for Runtime {
-		fn generate_proof(request: RuntimeApiDipProofRequest) -> Result<CompleteMerkleProof<Hash, DidMerkleProofOf<Runtime>>, RuntimeApiDipProofError> {
+	impl kilt_runtime_api_dip_provider::DipProvider<Block, runtime_api::DipProofRequest, CompleteMerkleProof<Hash, DidMerkleProofOf<Runtime>>, runtime_api::DipProofError> for Runtime {
+		fn generate_proof(request: runtime_api::DipProofRequest) -> Result<CompleteMerkleProof<Hash, DidMerkleProofOf<Runtime>>, runtime_api::DipProofError> {
 			let linked_did_info = match <Runtime as pallet_dip_provider::Config>::IdentityProvider::retrieve(&request.identifier) {
 				Ok(Some(linked_did_info)) => Ok(linked_did_info),
-				Ok(None) => Err(RuntimeApiDipProofError::IdentityNotFound),
-				Err(e) => Err(RuntimeApiDipProofError::IdentityProviderError(e))
+				Ok(None) => Err(runtime_api::DipProofError::IdentityNotFound),
+				Err(e) => Err(runtime_api::DipProofError::IdentityProviderError(e))
 			}?;
-			DidMerkleRootGenerator::<Runtime>::generate_proof(&linked_did_info, request.version, request.keys.iter(), request.should_include_web3_name, request.accounts.iter()).map_err(RuntimeApiDipProofError::MerkleProofError)
+			DidMerkleRootGenerator::<Runtime>::generate_proof(&linked_did_info, request.version, request.keys.iter(), request.should_include_web3_name, request.accounts.iter()).map_err(runtime_api::DipProofError::MerkleProofError)
```

Comment:
These lines look very complicated. Can we maybe split it? Especially if it is used as template

Reply:
Like this? https://github.com/KILTprotocol/kilt-node/pull/574/commits/24cb5ac5c19b80461a0d86ba6515b8c521aa255a

Reply:
Yes, I think it is already better. Maybe we can have a type `IdentityProviderOf<Runtime>` instead of `<Runtime as pallet_dip_provider::Config>::IdentityProvider`. But this is not really urgent or important.

Reply:
Yup: https://github.com/KILTprotocol/kilt-node/pull/574/commits/f73778daaf1dbf38db5d0a6a948698c124001146.